### PR TITLE
Refactor and improve tests concerning the TLE file handling

### LINKED ISSUE #119: tlefile.py seem to use satellite name rather than the international designator number
#### Code Sample, a minimal, complete, and verifiable piece of code

I have two simple tle files like this:
```
cat jpss2.tle
NOAA 21 (JPSS-2)        
1 54234U 22150A   23045.56664999  .00000332  00000+0  17829-3 0  9993
2 54234  98.7059 345.5113 0001226  81.6523 278.4792 14.19543871 13653
```
and
```
cat noaa21.tle 
NOAA 21
1 54234U 22150A   23045.56664999  .00000332  00000+0  17829-3 0  9993
2 54234  98.7059 345.5113 0001226  81.6523 278.4792 14.19543871 13653
```

And a `platforms.txt` file that contains NOAA-21, like this (in the latest Pyorbital release this files does not include NOAA-21):
```
...
NOAA-19 33591
NOAA-20 43013
NOAA-21 54234
...

```

```python
# Your code here
from pyorbital import tlefile
tle1 = tlefile.read('noaa 21', '/path/to/tles/noaa21.tle')
tle2 = tlefile.read('noaa 21', '/path/to/tles/jpss2.tle'
```
#### Problem description

The above code fails giving a KeyError, see below for a traceback.

I would have expected that when reading the TLE file the name of the satellite should not be used, but that Pyorbital exclusively goes by the designator number.
Worth mentioning that having a name consistent with what is listed in `platforms.txt` works (a dash in the platform name):
`tle2 = tlefile.read('noaa-21', '/path/to/tles/jpss2.tle'`

Of relevance, the same error appears when reading the TLE file over internet (Spacetrack):
```
from pyorbital.orbital import Orbital
orb = Orbital("NOAA 21")
```
But doing `orb = Orbital("NOAA-21")` works. Consistent with the above.

[this should also explain **why** the current behaviour is a problem and why the 
expected output is a better solution.]

#### Expected Output
The above code should run without errors/exceptions

#### Actual Result, Traceback if applicable
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[3], line 1
----> 1 tle = tlefile.read('noaa 21', '/home/a000680/data/tles/jpss2.tle')

File ~/usr/src/forks/pyorbital/pyorbital/tlefile.py:113, in read(platform, tle_file, line1, line2)
    105 def read(platform, tle_file=None, line1=None, line2=None):
    106     """Read TLE for *platform*.
    107 
    108     The data are read from *tle_file*, from *line1* and *line2*, from
   (...)
    111 
    112     """
--> 113     return Tle(platform, tle_file=tle_file, line1=line1, line2=line2)

File ~/usr/src/forks/pyorbital/pyorbital/tlefile.py:159, in Tle.__init__(self, platform, tle_file, line1, line2)
    156 self.mean_motion = None
    157 self.orbit = None
--> 159 self._read_tle()
    160 self._checksum()
    161 self._parse_tle()

File ~/usr/src/forks/pyorbital/pyorbital/tlefile.py:200, in Tle._read_tle(self)
    197     tle = _get_first_tle(uris, open_func, platform=self._platform)
    199     if not tle:
--> 200         raise KeyError("Found no TLE entry for '%s'" % self._platform)
    202 self._line1, self._line2 = tle.split('\n')

KeyError: "Found no TLE entry for 'NOAA 21'"
```

#### Versions of Python, package at hand and relevant dependencies

Thank you for reporting an issue !

### PR DESCRIPTION
**Pull request by adybbroe**
This PR was originally started to introduce support for NOAA-21.

The PR started out attempting to address an issue found when the user requests TLEs for a name not matching what is in the `platforms.txt` file and only partially matching what name is written in the TLE files. In particular this was a bit confusing when NOAA-21 was newly introduced and the name was `NOAA 21 (JPSS-2)`in the TLE files from Space_Track/Celetrak, while for NOAA-20 the name was `NOAA 20` in the same files.

Support for NOAA-21 was actually already included in #113 but that PR is about environment variables.

 - [x] Closes #119  
 - [x] Tests added 
 - [ ] Fully documented

### CODE REVIEW

**Comment by adybbroe** [comment]
While working with this PR and upgrading our systems for NOAA-21 at SMHI we realized that there seem to be a bug here, standing in the way of updating support for NOAA-21, see #119 
So, perhaps this PR should try fix this as well?

File: pyorbital/tests/test_tlefile.py [hunk 1]
```diff
@@ -48,8 +55,53 @@
...
+
+@pytest.fixture
+def fake_tlefile(tmp_path: Path) -> Path:
+    """Make fake tle file."""
+    filename = tmp_path / 'sometlefile.txt'
+    _write_fake_tle_file(filename)
+    yield filename
+
+
+line1_2 = "1 38771U 12049A   21137.30264622  .00000000  00000+0 -49996-5 0 00017"
```
Comment (adybbroe) [inline pyorbital/tests/test_tlefile.py:103]:
```suggestion
```

Comment (adybbroe) [inline pyorbital/tests/test_tlefile.py:103]:
```suggestion
```

File: pyorbital/tests/test_tlefile.py [hunk 2]
```diff
@@ -48,8 +55,53 @@
...
+line2_2 = "2 38771  98.7162 197.7716 0002383 106.1049 122.6344 14.21477797449453"
```
Comment (adybbroe) [inline pyorbital/tests/test_tlefile.py:104]:
```suggestion
```

File: pyorbital/tests/test_tlefile.py [hunk 3]
```diff
@@ -48,8 +55,53 @@
 LINE1 = "1 25544U 98067A   08264.51782528 -.00002182  00000-0 -11606-4 0  2927"
 LINE2 = "2 25544  51.6416 247.4627 0006703 130.5360 325.0288 15.72125391563537"
```
Comment (adybbroe) [inline pyorbital/tests/test_tlefile.py:57]:
```suggestion
LINE1_2 = "1 38771U 12049A   21137.30264622  .00000000  00000+0 -49996-5 0 00017"
LINE2_2 = "2 38771  98.7162 197.7716 0002383 106.1049 122.6344 14.21477797449453"
```

**Comment by adybbroe** [comment]
I have come to the conclusion that there is really no serious issue easily solved as otherwise suggested in #119 
Either the user specify a WMO Oscar based satellite name matching what is in the `platforms.txt`, so `NOAA-21` (already in main) or the user must specify an exact match to what is written in the TLE files. So if the TLE files look like this:

```
NOAA 21 (JPSS-2)        
1 54234U 22150A   23045.56664999  .00000332  00000+0  17829-3 0  9993
2 54234  98.7059 345.5113 0001226  81.6523 278.4792 14.19543871 13653
```

The use must specify `NOAA 21 (JPSS-2)` as the platform name, as in:
`tle_jpss2 = tlefile.read('NOAA 21 (JPSS-2)', './jpss2.tle')`

The only thing this PR does now is to add a debug message indicating that there might be a match in the TLE file for the the satellite the user is wanting the TLEs for. So, like:
```
>>> tle_metop = tlefile.read('Metop', '/data/tles/tle-202409060015.txt')
[DEBUG: 2024-11-23 14:30:03 : pyorbital.tlefile] Path to the Pyorbital configuration (where e.g. platforms.txt is found): /home/a000680/usr/src/forks/pyorbital/pyorbital/etc
[DEBUG: 2024-11-23 14:30:03 : pyorbital.tlefile] Found a possible match: METOP-B?
[DEBUG: 2024-11-23 14:30:03 : pyorbital.tlefile] Found a possible match: METOP-C?
```
And:
```
>>> tle_n21 = tlefile.read('NOAA 21', './jpss2.tle')
[DEBUG: 2024-11-23 14:38:15 : pyorbital.tlefile] Path to the Pyorbital configuration (where e.g. platforms.txt is found): /home/a000680/usr/src/forks/pyorbital/pyorbital/etc
[DEBUG: 2024-11-23 14:38:15 : pyorbital.tlefile] Found a possible match: NOAA 21 (JPSS-2)?
```

It means that in case the user gives a name like "NOAA" or "Metop" or even "N" or something far from a real satellite name, there might be several debug messages with more or less useful suggestions. But I think that can be accepted.

Okay?

File: pyorbital/tlefile.py [hunk 4]
```diff
@@ -343,41 +336,60 @@ def _get_first_tle(uris, open_func, platform=""):
...
+        tle = _merge_tle_from_two_lines(l_1, l_2)
+    elif l_0.strip().startswith(designator):
+        if (platform in SATELLITES or not only_first) or open_is_dummy:
+            l_1 = l_0
+            l_2 = _decode(next(fid))
+            tle = _merge_tle_from_two_lines(l_1, l_2)
+            if platform:
+                LOGGER.debug("Found platform %s, ID: %s", platform, SATELLITES[platform])
+    elif l_0.startswith(platform) and platform not in SATELLITES:
+        LOGGER.debug("Found a possible match: %s?", str(l_0.strip()))
```
Comment (pnuu) [inline pyorbital/tlefile.py:383]: I'm wondering should there be something other than just the log message here? The tests pass, but :man_shrugging:

**Review by pnuu — CHANGES_REQUESTED** [review]
Umm, this PR currently actually does only do refactoring and not add the mentioned NOAA-21 to the platforms.txt. So `git commit` missing @adybbroe ?

**Comment by adybbroe** [comment]
> Umm, this PR currently actually does only do refactoring and not add the mentioned NOAA-21 to the platforms.txt. So `git commit` missing @adybbroe ?

Yeah right, after a long discussion (mainly with myself) it ended up differently from what I anticipated when I started. Should we rename it then? I am okay with that!

**Comment by adybbroe** [comment]
> Umm, this PR currently actually does only do refactoring and not add the mentioned NOAA-21 to the platforms.txt. So `git commit` missing @adybbroe ?

Better now @pnuu ?

File: pyorbital/tlefile.py (hunk 4 above)
Comment (pnuu) [reply pyorbital/tlefile.py:383]: Nope, just wanted to solve the pre-commit complaint and afterwards got confused about the former title and description of the issue :sweat_smile:

**Comment by pnuu** [comment]
> Better now @pnuu ?

Much better! Now the title and description match what's being changed :+1:

**Review by mraspaud — COMMENTED** [review]
LGTM!

**Pull request merged**